packages/storybook/cxl-ui/cxl-card/[theme=cxl-course].stories.js

```diff
@@ -0,0 +1,77 @@
+import { html } from 'lit';
+import '@conversionxl/cxl-ui/src/components/cxl-card.js';
+import '@conversionxl/cxl-lumo-styles';
+
+export default {
+  title: 'CXL UI/cxl-card',
+  parameters: {
+    layout: 'centered',
+  },
+};
+
+const Template = ({ el, State }) => {
+  const stateArray = Array.isArray(State) ? State : [State];
+
+  return html`
+    <cxl-card
+      id="${el.cxl_hybrid_attr_post['@attributes'].id}"
+      class="${el.cxl_hybrid_attr_post['@attributes'].class}"
+      theme="cxl-course ${stateArray.map(state => `cxl-course-${state}`).join(' ')}"
+    >
+      <header class="entry-header">
+        ${stateArray.map(state =>
+          state === 'new'
+          ? html`<vaadin-icon icon="cxl:new"></vaadin-icon>`
+          : html``
+        )}
+        <div>
+          <div class="entry-title">
+            Account based marketing
+            ${stateArray.map(state =>
+              state === 'completed'
+              ? html`<vaadin-icon icon="lumo:checkmark"></vaadin-icon>`
+              : html``
+            )}
+          </div>
+          <div class="entry-byline">
+            <div><vaadin-icon icon="lumo:clock"></vaadin-icon>3h 00min</div>
+            <div>By: Ton Wesseling</div>
+          </div>
+          ${stateArray.map(state =>
+            state === 'with-badges'
+            ? html`
+              <div class="entry-summary">
+                <span theme="badge primary pill">Team Roadmap</span>
+                <span theme="badge secondary pill">Personal Roadmap</span>
+              </div>`
+            : html``
+          )}
+        </div>
```

Comment:
I'm concerned about the logic being present in storybook instead of encapsulating it into the component.

As I see, adding state param eg. `cxl-course-new` will not make new icon visible. It'll just apply specific styles, but if icon is not present... card will look the same as without this state.

For completed the same: we need to manually put an icon instead of just adding `cxl-course-completed` theme element.

Which makes me think: why do we even need to define states?

@lkraav what's your opinion?

Reply:
With badges it should just work out of the box: put badges and they're displayed nicely without a need to add theme name.

Reply:
> As I see, adding state param eg. `cxl-course-new` will not make new icon visible. It'll just apply specific styles, but if icon is not present... card will look the same as without this state.
> 
> For completed the same: we need to manually put an icon instead of just adding `cxl-course-completed` theme element.
> 
> Which makes me think: why do we even need to define states?

I think I agree - if possible, we should aim for an atomic (single) operation to achieve our goal. "Need to know multiple steps" is road to confusion.

Reply:
Alright, @kertuilves please encapsulate these states in the component.

`new` -> icon appears (without a need to put icon manually)

`completed` -> checkmark (without a need to put icon manually)

`with-badges` -> drop state; badges need to look good without need to add additional theme, just `cxl-course`